client/roomlistdock.cpp

```diff
@@ -194,10 +195,14 @@ RoomListDock::RoomListDock(MainWindow* parent)
     forgetAction =
         roomContextMenu->addAction(QIcon::fromTheme("irc-remove-operator"),
         tr("Forget room"), this, [this] {
-            if (auto room = getSelectedRoom())
-            {
-                Q_ASSERT(room->connection());
-                room->connection()->forgetRoom(room->id());
+            if (auto room = getSelectedRoom()) {
+                QMessageBox::StandardButton confirmation = QMessageBox::question(
+                    this, tr("Forget this room?"),
+                    tr("Are you sure you want to forget room %1?").arg(room->name()));
```

Comment:
```suggestion
                    tr("Are you sure you want to forget room %1?").arg(room->displayName()));
```
`room->name()` may return an empty string if the room has no set name; `displayName()`, on the other hand, tries really hard to produce something human-friendly, trying the room name, canonical alias, and even listing a few members if everything else fails.